lib/storage/providers/IDBKeyValProvider.js

```diff
@@ -4,30 +4,69 @@ import utils from '../../utils';
 
 // We don't want to initialize the store while the JS bundle loads as idb-keyval will try to use global.indexedDB
 // which might not be available in certain environments that load the bundle (e.g. electron main process).
-let customStoreInstance;
-const getCustomStore = () => {
-    if (!customStoreInstance) {
-        customStoreInstance = createStore('OnyxDB', 'keyvaluepairs');
-    }
-    return customStoreInstance;
-};
+let idbKeyValStore;
 
 const provider = {
     /**
-     * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+     * The name of the provider that can be printed to the logs
+     */
+    name: 'IDBKeyValProvider',
+
+    /**
+     * Initializes the storage provider
+     */
+    init() {
+        const newIdbKeyValStore = createStore('OnyxDB', 'keyvaluepairs');
+
+        if (newIdbKeyValStore == null) throw Error('IDBKeyVal store could not be created');
```

Comment:
What problem specifically does this solve? Do we have any documented cases where `createStore()` returned `null`? What was the problem we saw? What about when an irrecoverable error happens when trying to save after initializing the store successfully? 

(that second situation is why this issue was created in the first place - and I'm not convinced we have solved it)